### PR TITLE
Normalize xunit.runner.visualstudio references

### PR DESCRIPTION
**Pull request by agocke**
The C# syntax tests were missing a reference to xunit.runner.visualstudio
which prevented them from working in the Test Explorer in 15.8 Preview.
Rather than just adding a reference to this one test, I've added a reference
to the root test utilities project, that almost every test references, and
deleted the explicit references in the test leaf projects. This should
hopefully be a more robust method of using the XUnit VS test adapter.

### CODE REVIEW

**Comment by agocke** [comment]
cc @fayrose This should fix the testing problem.

**Comment by jcouv** [comment]
I take it that all our test projects have a reference to `TestUtilities.csproj`. Is that the case?

**Comment by agocke** [comment]
@jcouv Yes, directly or indirectly. If the project didn't, I didn't remove the reference.

File: src/Test/Utilities/Portable/TestUtilities.csproj [hunk 1]
```diff
@@ -95,6 +95,7 @@
     <PackageReference Include="xunit" Version="$(xunitVersion)" />
     <PackageReference Include="xunit.analyzers" Version="$(xunitanalyzersVersion)" />
     <PackageReference Include="xunit.runner.console" Version="$(xunitrunnerconsoleVersion)" />
+    <PackageReference Include="xunit.runner.visualstudio" Version="$(xunitrunnervisualstudioVersion)" />
```
Comment (sharwell) [inline src/Test/Utilities/Portable/TestUtilities.csproj]: ❓ Why is this added? This project does not contain tests.

Comment (sharwell) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: Oh, I was expecting the reference to be in one of our shared **.targets** files, not here...

Comment (tmat) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: Let's add this to `build\Targets\Roslyn.Toolsets.Xunit.targets`. This targets file is imported to all unit test projects.

Comment (tmat) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: While you're changing this please also move `xunit` and `xunit.runner.console` there.
Also, `xunit.analyzers` do not need to be explicitly reference since they are referenced by `xunit`.

Comment (sharwell) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: > Also, xunit.analyzers do not need to be explicitly referenced since they are referenced by xunit.

We reference a newer version of the analyzers than would be brought in automatically, so the reference is still needed.

Comment (tmat) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: Ok, then let's move that reference to `build\Targets\Roslyn.Toolsets.Xunit.targets` as well

Comment (agocke) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: I disagree. We've made an explicit decision in the past to put all our package references in project files, not target files. This is for a couple reasons, but mainly it's to prevent references from being magically included or excluded based on target configurations. I think we should put these in the project files, like we do with all of our other package references.

Comment (agocke) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: @jaredpar I believe you had strong feelings on this. Care to weigh in?

Comment (agocke) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: I tried this and it doesn't work. The "TestUtilities" projects don't import the Roslyn.Toolsets.Xunit.targets file because the targets file is only for actual unit test projects. Technically, the runner isn't necessary in the utilities projects either, so just the runner references could be added to the targets file, but this means that half the xunit references are in project files and half the references are in targets files. This is more confusing than either choice. I think we should stick with the original design.

Comment (tmat) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: The runner should not be referenced by non-test projects. TestUtilities are not a test project. They should only reference the xunit packages they need (xunit.abstractions i think).

Comment (agocke) [reply src/Test/Utilities/Portable/TestUtilities.csproj]: The console runner is already referenced. This works fine.
